plugins/hls-change-type-signature-plugin/test/Main.hs

```diff
@@ -1,3 +1,8 @@
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 908
+{-# OPTIONS_GHC -Wwarn=x-partial #-}
+#endif
+
```

Comment:
I went for this to make the disabling as local as possible.
But since you're proponent of using partial functions in tests, I think it makes sense to use this cabal file variant in  specific test suite targets that use partial functions.